### ISSUE TITLE
Neo eddits + Workflow rendering Neo blocks in the wrong tree.

### ISSUE THREAD

**Issue by iainhenderson** [engram-design/Workflow#135]
My apologies for the vagueness of this issue, reproducing the problem consistently has been difficult.

After a recent round of plugin and CMS updates (Including Craft 3.6.10 > 3.7.11, Workflow 1.5.0 > 1.6.2, Neo 2.10.0 > 2.11.9) my client has reported occasional problems with a Neo's structure becoming corrupted, with it attempting to render blocks in parent blocks which don't allow, or support it causing template errors.

After spending time trying to replicate the issue, the only way I managed to recreate it was by running through the Workflow process as Editor/Publisher. When submitting drafts / and approving them I noticed this error showing in the JS console:

![Screen Shot 2021-09-10 at 1 17 47 pm](https://user-images.githubusercontent.com/775758/133010330-ff32c56f-af04-46be-97ff-f16e9dc3c74c.png)

After the issue was noted we updated from Craft 3.7.11 > 3.7.12 and Neo 2.11.9 > 2.11.12 as they mentioned improvements working with drafts, hoping this would solve the issue. It didn't.

While browsing recent Workflow issues I found https://github.com/verbb/workflow/issues/134 . We are also running Blitz 3.10.3, I haven't managed to reproduce the issue after disabling Blitz. I  also tried re-enabling Blitz and rolling the Workflow back to 1.6.1 and couldn't recreate the issue in that state either. Although as mentioned earlier, I don't have a method to consistently trigger the issue yet, so I have no idea if these things are actually fixes.

My current reproduction process is this:
1. As an editor, Make edits to a Neo field - adding new blocks and rearranging child/parent blocks
2. As editor, comment and submit for review
3. As Publisher, approve the edit
4. As Author reopen entry, live preview and see errors. See Neo fields in the wrong tree. 

**Additional info**
- Plugin version: 1.6.2
- Craft version: 3.7.12
- Neo version: 2.11.12
- Blitz version: 3.10.3
- Multi-site: No

**Comment by leightonhughes** [comment]
Hey @iainhenderson, think your issue is related to mine #134. Did you get anywhere in understanding what is happening? My client is approving and it just hangs and doesn't go through. Completely stumped on how to resolve.
<img width="396" alt="CleanShot 2021-09-20 at 12 43 51@2x" src="https://user-images.githubusercontent.com/5158786/133952691-d12a62c2-00d1-45fb-9778-97dcbb2048de.png">

**Comment by iainhenderson** [comment]
Hi @leightonhughes, I looked at your issue before opening mine. I initially thought they might be linked as well as both problems seem to recent developments and we were using the same Workflow and Blitz versions. The issue itself seems to be different though, my submissions/approvals don't fail (although it is sometimes quite slow) My issue is Worklfow seems to be occasionally messing up the Neo field.

Frustrating I still haven't nailed down a way to reliably reproduce them problem. Out of  about 40 attempts I've triggered it twice. This means I don't know if disabling Blitz or backtracking to an older version of workflow actually had any effect, or if just didn't pop up while testing those changes.

The JS error I posted seems to be a Craft issue, not related directly unrelated to Workflow. It should be fixed next release: See https://github.com/craftcms/cms/issues/9841 . Maybe with the changes in   https://github.com/craftcms/cms/commit/1663031821d09ea5442b2169a0518e5b0115f342 we might be able to learn more about why saving the draft is failing.

So no, in short I'm no closer to figure out what's going on.
